File: Swift/SelectionSort.swift
import Foundation
func selectionSort(array: inout [Int]) {
    for currentIndex in 0..<(array.count - 1) {
        var minIndex = currentIndex;
        for i in (currentIndex + 1)..<array.count {
            if (array[i] < array[minIndex]) {
                minIndex = i;
            }
        }
        if (minIndex != currentIndex) {
            let temp = array[currentIndex];
            array[currentIndex] = array[minIndex];
            array[minIndex] = temp;
        }
    }
}

var array = [12, 11, 15, 10, 9, 1, 2, 3, 13, 14, 4, 5, 6, 7, 8]
selectionSort(array: &array)
print(array)

File: C/Selection-Sort/SelectionSort.c
#include "SelectionSort.h"

int main(void)
{
    int SIZE;
    do
    {
        printf("how many  array's elements you want: ");
        scanf("%d", &SIZE);
    } while(SIZE > 20); // max = 20

    // memory allocation for array in heap segement
    int *array = malloc(SIZE * sizeof(int));
    array_init(array, SIZE); 
    array_print(array, SIZE); // before sorting
    selection_sort(array, SIZE); // sorting
    array_print(array, SIZE); // after sorting
    free(array); // deallocating the allocated virtual memory in heap segement
    return 0;
}
void array_init(int *array, int SIZE)
{
    printf("add values to your array:\n");
    for(int i = 0 ; i < SIZE ; i++)
    {
        printf("array[%d]: ", i);
        scanf("%d", array+i);
    }

}

void array_print(int *array, int SIZE)
{
    for(int i = 0 ; i < SIZE ; i++)
    {
        printf("%d ", *(array+i));
    }
    printf("\n");
}

void swap(int *a, int *b)
{
    int temp = *b;
    *b = *a;
    *a = temp; 
}

void selection_sort(int *array, int SIZE)
{
    for(int j = 0 ; j < SIZE ; j++)    
    {
        for(int i = j+1, min_index = j; i < SIZE ; i++)
        {            
            if(array[i] < array[min_index])
            {
                swap(array+i, array+min_index);
                min_index++;
            }
        }
    }
}